### PR TITLE
Handle CrabV2 UI for migration cases

### PR DESCRIPTION
**Pull request by KMKoushik**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] Added video recordings if it is a UI change

## User Facing Checklist

- [ ] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

File: packages/frontend/pages/strategies.tsx [hunk 1]
```diff
@@ -291,14 +289,16 @@ const Strategies: React.FC = () => {
...
-                { displayCrabV1 ? <CrabStrategyHistory /> : <CrabStrategyV2History /> }
+                {displayCrabV1 ? <CrabStrategyHistory /> : <CrabStrategyV2History />}
               </div>
               {supportedNetwork && (
                 <div className={classes.tradeCard}>
-                  {displayCrabV1 && !currentEthValue.isZero() && !isQueued? <MigrationNotice /> : null}
                   <div className={classes.tradeForm}>
                     {!!address ? (
-                      <CrabTradeComponent maxCap={maxCap} depositedAmount={vault?.collateralAmount || new BigNumber(0)} />
+                      <CrabTradeComponent
```
Comment (daryakaviani) [inline packages/frontend/pages/strategies.tsx:298]: Do we want to make a new `CrabV2TradeComponent`?

File: packages/frontend/src/components/Strategies/Crab/CrabTrade.tsx [hunk 2]
```diff
@@ -274,12 +274,14 @@ const CrabTrade: React.FC<CrabTradeType> = ({ maxCap, depositedAmount }) => {
           )}
           <div className={classes.tradeContainer}>
             {depositOption === 0 ? (
-              !isQueued ? (
-                <CrabMigration />
+              !currentEthValue.isZero() ? (
+                <Typography variant="body2" color="textSecondary" style={{ marginTop: '20px' }}>
+                  Migrate to V2 coming soon!
```
Comment (daryakaviani) [inline packages/frontend/src/components/Strategies/Crab/CrabTrade.tsx:279]: Maybe we should clarify that this is migration for those who didn't migrate during early access?

Something like:
> Didn't migrate from Crab V1 during early access? Migration to V2 coming soon.

File: packages/frontend/src/components/Strategies/Crab/CrabTrade.tsx [hunk 3]
```diff
@@ -274,12 +274,14 @@ const CrabTrade: React.FC<CrabTradeType> = ({ maxCap, depositedAmount }) => {
...
+                </Typography>
               ) : (
                 <>
                   <Typography variant="body2" color="textSecondary" style={{ marginTop: '8px' }}>
-                    You have secured your spot! Your position will be included in Crab v2 at launch 🦀🎉
+                    You don't have any Crab v1 position to migrate
```
Comment (daryakaviani) [inline packages/frontend/src/components/Strategies/Crab/CrabTrade.tsx:284]: Should be "You don't have **a** Crab v1 position to migrate"

File: packages/frontend/src/components/Strategies/Crab/CrabTradeV2.tsx [hunk 4]
```diff
@@ -267,6 +268,18 @@ const CrabTradeV2: React.FC<CrabTradeV2Type> = ({ maxCap, depositedAmount }) =>
             />
           </div>
           <div className={classes.tradeContainer}>
+            <div style={{ marginBottom: '8px' }}>
+              {isClaimAndWithdraw && currentEthActualValue.gt(0) ? (
+                <>
+                  <Typography variant="caption" component="div">
+                    - Withdraw initially transferred crab position
```
Comment (daryakaviani) [inline packages/frontend/src/components/Strategies/Crab/CrabTradeV2.tsx:275]: Not clear what "initially transferred" means. Migrated?

File: packages/frontend/src/state/crab/hooks.ts [hunk 5]
```diff
@@ -547,6 +559,34 @@ export const useFlashWithdrawV2 = () => {
...
+  const handleTransaction = useHandleTransaction()
+  const address = useAtomValue(addressAtom)
+  const calculateEthWillingToPay = useCalculateEthWillingToPayV2()
+
+  const claimAndWithdraw = useCallback(
+    async (amount: BigNumber, slippage: number, onTxConfirmed?: () => void) => {
+      if (!contract) return
+
+      const { maximumAmountIn: _ethWillingToPay } = await calculateEthWillingToPay(amount, slippage)
+      console.log(_ethWillingToPay.toString())
```
Comment (daryakaviani) [inline packages/frontend/src/state/crab/hooks.ts:573]: Can delete

File: packages/frontend/src/state/crab/hooks.ts [hunk 6]
```diff
@@ -589,6 +629,27 @@ export const useFlashWithdrawEthV2 = () => {
...
+  const { value: userCrabBalance } = useTokenBalance(crabStrategy2, 5, 18)
+  const contract = useAtomValue(crabStrategyContractAtomV2)
+  const claimAndWithdraw = useClaimWithdrawV2()
+
+  const claimAndWithdrawEth = useCallback(
+    async (ethAmount: BigNumber, slippage: number, onTxConfirmed?: () => void) => {
+      if (!contract) return
+
+      const equivalentCrab = ethAmount.div(currentEthValue).times(userCrabBalance)
+      console.log(currentEthValue?.toString(), userCrabBalance?.toString(), ethAmount.toString())
```
Comment (daryakaviani) [inline packages/frontend/src/state/crab/hooks.ts:644]: Can delete

File: packages/frontend/pages/strategies.tsx [hunk 7]
```diff
@@ -115,44 +116,48 @@ const useStyles = makeStyles((theme) =>
...
+  const CrabTradeComponent = displayCrabV1 ? CrabTrade : CrabTradeV2
+
+  useEffect(() => {
+    if (displayCrabV1) setStrategyData()
+  }, [collatRatio, displayCrabV1, setStrategyData])
 
   useEffect(() => {
-    setStrategyData()
-  }, [collatRatio, setStrategyData])
+    if (!displayCrabV1) setStrategyDataV2()
```
Comment (daryakaviani) [inline packages/frontend/pages/strategies.tsx:159]: Is the PnL right for V1? For V2, I had to call `setStrategyDataV2()` to have it update properly. But it looks reasonable for V2 here.
Screen Shot 2022-07-25 at 10 30 21 AM

**Review by daryakaviani — CHANGES_REQUESTED** [review]
Some minor comments. 

Two things that I think we should fix:

1. Super glitchy PnL. The PnL shows up before all atoms are properly loaded, so it glitches with a negative value for a few seconds. As you can see, both crab positions glitch to super negative values before settling at the right PnL.
https://user-images.githubusercontent.com/29926727/180841437-b76777c2-0b84-4a42-8c84-deb1539e043a.mov

2. Other positions do not show up. Around 50% of the time, non-strategy positions do not show up. Here, you can see me refreshing the page several times and only seeing my long position sometimes.
https://user-images.githubusercontent.com/29926727/180842358-e90002f2-779c-47cd-af44-3252e98ad75a.mov

File: packages/frontend/pages/strategies.tsx (hunk 7 above)
Comment (KMKoushik) [reply packages/frontend/pages/strategies.tsx:159]: Based on the displayCrabV1 it either calls setStrategyDataV2 or setStrategyData. So should not be a problem

**Pull request merged**